client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -18,10 +19,47 @@ import EditorMediaModalDetailPreviewVideo from './detail-preview-video';
 import EditorMediaModalDetailPreviewAudio from './detail-preview-audio';
 import EditorMediaModalDetailPreviewDocument from './detail-preview-document';
 import Button from 'components/button';
-import { userCan, isJetpack } from 'lib/site/utils';
+import { userCan } from 'lib/site/utils';
+import versionCompare from 'lib/version-compare';
 import MediaUtils, { isItemBeingUploaded } from 'lib/media/utils';
 import config from 'config';
 
+/**
+ * This function return true if the image editor can be
+ * enabled/shown
+ *
+ * @param  {object} item - media item
+ * @param  {object} site - current site
+ * @return {boolean} `true` is the image-editor can be enabled.
+ */
+const enableImageEdition = ( item, site ) => {
+	// do not allow if, for some reason, there isn't a valid item yet
+	if ( ! item ) {
+		return false;
+	}
+
+	// do not show if the feature flag isn't set
+	if ( ! config.isEnabled( 'post-editor/image-editor' ) ) {
+		return false;
+	}
+
+	// do not allow for private sites
+	if ( site.is_private ) {
+		return false;
+	}
+
+	// do not allow for Jetpack site with a non-valid version
+	const isJetpack = get( site, 'jetpack', false );
+	const jetpackVersion = get( site, 'options.jetpack_version', '0.0' );
+	const isValidVersion = versionCompare( jetpackVersion, '4.7-alpha', '>=' );
+
+	if ( isJetpack && ! isValidVersion ) {
```

Comment:
The above are going to get computed for non-jetpack sites which is bit of a waste.
I would:

```
if (
    get( site, 'jetpack', false ) &&
    versionCompare( get( site, 'options.jetpack_version', '0.0' ) , '4.7-alpha', '>=' )
) {
```

That is not a big clause and we dont have to compute versions for the vast majority of sites that are in calypso